lib/string_reverse.rb

```diff
@@ -1,4 +1,16 @@
 # A method to reverse a string in place.
 def string_reverse(my_string)
-  my_string << "not implemented"
+  #my_string << "not implemented"
+    index = 0
+    while my_string != nil && my_string.length != 0 && index < (my_string.length/2)

```

Comment:
You don't need to check for my_string being nil or of being empty for every iteration through the loop. Instead, you could have those as a check in a conditional statement before the loop begins like so:
```ruby
if my_string == nil || my_string.length == 0
   return
end
while index < (my_string.length/2)
...
```